account_commission_listing_fee/models/commission_settlement.py

```diff
@@ -0,0 +1,55 @@
+# Copyright 2022 Quartile Limited
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class CommissionSettlement(models.Model):
+    _inherit = "commission.settlement"
+
+    event_ids = fields.Many2many(
+        "event.event", compute="_compute_event_ids", string="Events", store=True
+    )
+    number_of_items = fields.Integer(compute="_compute_number_of_items", store=True)
+
+    @api.depends("line_ids")
+    def _compute_event_ids(self):
+        for settlement in self:
+            events = settlement.line_ids.product_id.event_ids
+            settlement.event_ids = events
+
+    @api.depends("event_ids", "agent_id")
+    def _compute_number_of_items(self):
+        for settlement in self:
+            event_ids = settlement.event_ids.ids
+            products = (
+                self.env["product.product"]
+                .search([("seller_id", "=", settlement.agent_id.id)])
+                .filtered(
+                    lambda x: set(x.event_ids.ids).issubset(event_ids)
+                    and x.event_ids.ids
+                )
+            )
```

Comment:
```suggestion
            event_ids = settlement.event_ids.ids
            if not event_ids:
                settlement.number_of_items = 0
                continue
            # We check if product.event_ids <= settlement.event_ids to avoid
            # counting the product more than once in multiple settlements.
            products = (
                self.env["product.product"]
                .search([("seller_id", "=", settlement.agent_id.id)])
                .filtered(lambda x: set(event_ids).issubset(x.event_ids.ids))
            )
```
My bad.  I think my suggestion was invalid.  Can you please double-check.

Reply:
I don't think we can use this search in product. The problem is if product A has one event and settlement has two event, number of items set to 0.
![Screenshot from 2022-12-13 21-23-58](https://user-images.githubusercontent.com/45355704/207367173-7c3f4435-8bd0-44f7-baa8-e378ba53dc64.png)

I think we can use like my lastest update.
```
products = (
                self.env["product.product"]
                .search([("seller_id", "=", settlement.agent_id.id)])
                .filtered(
                    lambda x: set(x.event_ids.ids).issubset(event_ids)
                    and x.event_ids.ids
                )
            )
```

Reply:
> I think we can use like my lastest update.
> 
> ```
> products = (
>                 self.env["product.product"]
>                 .search([("seller_id", "=", settlement.agent_id.id)])
>                 .filtered(
>                     lambda x: set(x.event_ids.ids).issubset(event_ids)
>                     and x.event_ids.ids
>                 )
>             )
> ```

Thanks for checking.  I think you are right. Can we adjust the filtered args like below though.

```python
                .filtered(lambda x: x.event_ids and set(x.event_ids.ids).issubset(event_ids))
```

Reply:
Yes. How about to avoid counting the product more than once in multiple settlements? Should we use indicator in product.product for checking counted product or not? I feel like this is invalid. Can you please suggest me?

Reply:
> Thanks for checking. I think you are right.

No, this is not correct as I check again.  We need to prevent the situation of a product being counted more than once.

> The problem is if product A has one event and settlement has two event, number of items set to 0.

This situation is not expected to happen, although we need to double check.  We may keep this remark in the code/readme as an assumption and implement a constraint as such, or come up with another approach such as flagging the product as it gets counted in a settlement (I suppose linking the settlement to products may do - if a product is already linked to an non-canceled settlement then this product should not be counted in a new settlement).  What do you think?

Reply:
> This situation is not expected to happen, although we need to double check.

Yes. But I think we need to cover for all situation. So, I prefer using filter like this `.filtered(lambda x: x.event_ids and set(x.event_ids.ids).issubset(event_ids))`.

> I suppose linking the settlement to products may do - if a product is already linked to an non-canceled settlement then this product should not be counted in a new settlement

I think that is a good choice. I will work on this one.